src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -691,8 +696,17 @@ private static void EmitSimplifiedGo(IndentedTextWriter writer, RegexMethod rm,
             bool rtl = code.RightToLeft;
             bool hasTimeout = false;
 
-            int localCounter = 0;
-            string NextLocalName(string prefix) => $"{prefix}{localCounter++}";
+            // Helper to define names.  Names start unadorned, but as soon as there's repetition,
+            // they begin to have a numbered suffix.
+            var usedNames = new Dictionary<string, int>();
+            string ReserveName(string prefix)
```

Comment:
More of a question than a comment, do we have any convention for inline functions and where should they be defined in a method for the repo? I've seen other repos define the convention of defining all after the return statement just to make it obvious they are inline funcitons but not sure if we should have something like that. I know we have a bunch of these in Regex, I guess this one particularly jumped out as we are defining it here but using it a few lines below.

Reply:
I don't think we have much of a convention at this point.  We can certainly move things around here if you have a preference.